File: 프로그래머스/JAVA/Lv.1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        String number = String.valueOf(x);
        int sum = 0;
        for(int i=0; i<number.length(); i++){
            sum+=Integer.valueOf(number.substring(i,i+1));
        }
        boolean answer = true;
        if(x%sum==0){
            answer = true;
        }else{
            answer = false;
        }
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.py
def solution(x):
    answer = True
    tmp = x
    num = 0
    while True:
        if tmp < 10:
            num += tmp
            break
        num += int(tmp%10)
        tmp = int(tmp/10)
    
    if x % num != 0:
        answer = False
        
    return answer